sdk/core/Azure.Core.TestFramework/src/Instrumentation/DiagnosticScopeValidatingInterceptor.cs

```diff
@@ -180,6 +181,10 @@ internal static async ValueTask<T> ValidateDiagnosticScope<T>(Func<ValueTask<(T
             {
                 // Remove subscribers before enumerating events.
                 diagnosticListener.Dispose();
+                bool? skipOverride = forwardAttribute is null
+                    ? default(bool?)
+                    : (bool?)forwardAttribute.GetType().GetProperty("SkipChecks").GetValue(forwardAttribute);
+                skipChecks |= skipOverride.HasValue && skipOverride.Value;
```

Comment:
Since we know the type of the property is bool, should we cast as (bool) and simplify the need to check HasValue?

Reply:
it can be null for the default constructor so if we call .Value or try to cast to bool we will get an exception

Reply:
Does it need to be nullable at all? The default value would be `false` and work the same way, I believe.

Reply:
I think the challenge is this needs to get merged with the global client level skipChecks.  I think we need to differentiate from "not set" and "explicit false" that way we only override the global client level setting when you explicitly set it

Reply:
I guess its not technically an override its an also.  Here is how i understand the expected behavior

(Class, Method)
(true, true) = skip checks
(true, false) = skip checks
(false, true) = skip checks
(false, false) = don't skip checks|

If we expand that to include bool? the results don't change since its not override the class `false` its only overriding the class `true`

Reply:
I may be missing something, but doesn't `skipChecks` evaluate to the same value regardless if the value is `skipOverride` is `null` vs `false`?

Reply:
```c#
bool skipChecks = false;
bool? skipOverride = null;
bool boolSkip = false;
bool nullResult = skipChecks |= skipOverride.HasValue && skipOverride.Value;
bool boolResult = skipChecks |= boolSkip;
Console.WriteLine(nullResult);  //False
Console.WriteLine(boolResult); //False
```

Reply:
yeah, if it was a method level override we need the null.  I think I am agreeing with you now that it isn't an override its an OR against the class level value.